packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts

```diff
@@ -1669,23 +1673,138 @@ export const transformPlotlyJsonToFunnelChartProps = (
 
 export const projectPolarToCartesian = (input: PlotlySchema): PlotlySchema => {
   const projection: PlotlySchema = { ...input };
+
+  // 1. Find the global min and max radius across all series
+  let minRadius = 0;
+  let maxRadius = 0;
+  for (let sindex = 0; sindex < input.data.length; sindex++) {
+    const rVals = (input.data[sindex] as Partial<PlotData>).r;
+    if (rVals && isArrayOrTypedArray(rVals)) {
+      for (let ptindex = 0; ptindex < rVals.length; ptindex++) {
+        if (!isInvalidValue(rVals[ptindex])) {
+          minRadius = Math.min(minRadius, rVals[ptindex] as number);
+          maxRadius = Math.max(maxRadius, rVals[ptindex] as number);
+        }
+      }
+    }
+  }
+
+  // 2. If there are negative radii, compute the shift
+  const radiusShift = minRadius < 0 ? -minRadius : 0;
+
+  // 3. Project all points and create a perfect square domain
+  const allX: number[] = [];
+  const allY: number[] = [];
+  let originX: number | null = null;
   for (let sindex = 0; sindex < input.data.length; sindex++) {
     const series = input.data[sindex] as Partial<PlotData>;
+    // If scatterpolar, set text to theta values as strings
+    if (series.type === 'scatterpolar' && Array.isArray(series.theta)) {
+      series.text = series.theta.map(v => String(v));
+    }
     series.x = [] as Datum[];
     series.y = [] as Datum[];
-    for (let ptindex = 0; ptindex < (series.r?.length ?? 0); ptindex++) {
-      if (isInvalidValue(series.theta?.[ptindex]) || isInvalidValue(series.r?.[ptindex])) {
+    const thetas = series.theta!;
+    const rVals = series.r!;
+
+    // Skip if rVals or thetas are not arrays
+    if (!isArrayOrTypedArray(rVals) || !isArrayOrTypedArray(thetas)) {
+      projection.data[sindex] = series;
+      continue;
+    }
+
+    // Compute tick positions if categorical
+    let uniqueTheta: Datum[] = [];
+    let categorical = false;
+    if (!isNumberArray(thetas)) {
+      uniqueTheta = Array.from(new Set(thetas));
+      categorical = true;
+    }
+
+    for (let ptindex = 0; ptindex < rVals.length; ptindex++) {
+      if (isInvalidValue(thetas?.[ptindex]) || isInvalidValue(rVals?.[ptindex])) {
         continue;
       }
 
-      const thetaRad = ((series.theta![ptindex] as number) * Math.PI) / 180;
-      const radius = series.r![ptindex] as number;
-      series.x.push(radius * Math.cos(thetaRad));
-      series.y.push(radius * Math.sin(thetaRad));
+      // 4. Map theta to angle in radians
+      let thetaRad;
+      if (categorical) {
+        const idx = uniqueTheta.indexOf(thetas[ptindex]);
+        const step = (2 * Math.PI) / uniqueTheta.length;
```

Comment:
Could you update the step logic to match with plotly. Angle 0 starts at (0, r) and goes in clockwise direction

Reply:
Lets take the comments in a follow up PR. Not blocking this commit.